lib/reek/smells/smell_warning.rb

```diff
@@ -88,25 +95,23 @@ def smell_classes
         [smell_detector.smell_category, smell_detector.smell_type]
       end
 
-      def common_parameters_equal?(other_parameters)
-        other_keys   = other_parameters.keys
-        other_values = other_parameters.values
+      def check_attributes_comparability(other_attributes)
+        other_attributes.keys.each do |attr|
+          unless COMPARABLE_ATTRIBUTES.include?(attr)
+            raise ArgumentError, "The attribute #{attr} you want to check for doesn't exist"
+          end
+        end
+      end
 
+      def common_parameters_equal?(other_parameters)
+        return true unless other_parameters
```

Comment:
This return is unnecessary (that's why it wasnt there before) and de facto a hidden type check, lets drop this line.